### PR TITLE
Segment Analytics (APPS-760, APPS-764, APPS-784)

### PR DESCRIPTION
**Pull request by manish2kcoder**
Combined PR for all tickets for Segment Analytics requirements.

**Disable Segment analytics by default**

    static let kSegmentAnalytics = false // if enabled,make sure kSegmentAccountID is having correct value
    static let kSegmentAccountID = "" // must have some value if kSegmentAnalytics is enabled

### CODE REVIEW

File: Zype/Helpers/AnalyticsProtocolDelegate.swift [hunk 1]
```diff
@@ -60,4 +61,77 @@ extension PlayerVC: AnalyticsHelperProtocol {
...
         return (beacon: configUrl, customDimensions: customDimensions)
     }
+
+    func getSegmentAnalyticsFromResponse(_ playerObject: VideoObjectModel?) -> [String: Any] {
+        var customDimensions = [String: Any]()
+        customDimensions[SegmentAnalyticsAttributes.contentCmsCategory.rawValue] = "null"
+        customDimensions[SegmentAnalyticsAttributes.adType.rawValue] = "null"
+        customDimensions[SegmentAnalyticsAttributes.contentShownOnPlatform.rawValue] = "ott"
+        customDimensions[SegmentAnalyticsAttributes.streaming_device.rawValue] = "Apple AppleTV"
+        customDimensions[SegmentAnalyticsAttributes.videoAccountId.rawValue] = "416418724"
```
Comment (imchrissmith) [inline Zype/Helpers/AnalyticsProtocolDelegate.swift]: This value should be inherited from the Const.swift file and not hardcoded: `static let kSegmentAccountID = "" // must have some value if kSegmentAnalytics is enabled`

File: Zype/Helpers/AnalyticsProtocolDelegate.swift [hunk 2]
```diff
@@ -60,4 +61,77 @@ extension PlayerVC: AnalyticsHelperProtocol {
...
+        customDimensions[SegmentAnalyticsAttributes.videoAccountName.rawValue] = "People"
```
Comment (imchrissmith) [inline Zype/Helpers/AnalyticsProtocolDelegate.swift]: This value should be pulled from the App Title, not hardcoded

File: Zype/Const.swift [hunk 3]
```diff
@@ -50,6 +50,9 @@ class Const: NSObject {
     static let kLiveItemEnabled = false
     static let kLiveVideoID = "5c8faa021d1f4314dd006203"
     static let kEpisodeNumberDisplay = false
+    static let kSegmentAnalytics = false // if enabled, make sure kSegmentAccountID has a value entered
+    static let kSegmentAnalyticsWriteKey = "enter_write_key_here" // must have some value if kSegmentAnalytics is enabled
```
Comment (imchrissmith) [inline Zype/Const.swift]: This value should not be hardcoded. It needs to be entered by the customer. I've removed it from the commit.

**Review by imchrissmith — CHANGES_REQUESTED** [review]
@manish2kcoder Please address the issues highlighted in this review and let me know once complete so we can review and merge.

**Review by imchrissmith — APPROVED** [review]

File: Zype/Helpers/AnalyticsProtocolDelegate.swift (hunk 2 above)
Comment (manish2kcoder) [reply Zype/Helpers/AnalyticsProtocolDelegate.swift]: 😞 got confused after looking at the APPS-784 description that it's hardcoded, that's fine for PeopleTv but this is base repo so it should not be hardcoded. Thanks for catching this in review.

**Pull request merged**